zerver/lib/outgoing_webhook.py

```diff
@@ -244,8 +244,17 @@ def notify_bot_owner(
     if failure_message:
         notification_message += "\n" + failure_message
     if status_code:
+        if status_code == 407:
+            notification_message += (
+                "\nThe URL configured for the webhook is for a private or disallowed network."
+            )
+        else:
+            notification_message += (
+                f"\nThe webhook got a response with status code *{status_code}*."
+            )
+    if response_content and status_code != 407:
         notification_message += f"\nThe webhook got a response with status code *{status_code}*."
```

Comment:
It seems like this is the same case as the one you have slightly above in the `else` clause -- what's your reasoning there?  

More generally I think you might be better off using an `else if` structure to make it more clear under what circumstances one might have multiple blocks.  E.g. probably you want an `if status_code == 407` as the top conditional here.

Reply:
> It seems like this is the same case as the one you have slightly above in the else clause -- what's your reasoning there?
ah! this is really embarassing! this does the same thing as the previous else clause it should not be here...
also the above `else` clause is there because as per my understanding we wanted to have that message only when `status_code == 407` ... please correct me if I am wrong...

I have restructured the conditions as follows:
```
if status_code == 407:
    notification_message += f"...."
elif status_code:
     notification_message += f"...."
if response_content and status_code != 407:
    notification_message += f"...."
```
the last `status_code != 407` is there because it was independent of the status_code, if we dont have the extra check for 407, then it will display the payload in the message which we want to avoid...